src/Flights.groovy

```diff
@@ -0,0 +1,108 @@
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+class Flights {
+    static void main(String[] args) {
+        def flights = []
+        createFlights(flights)
+        def passengersToSof = flights.each { it.destination == "SOF"}.collect {it.passengers}.sum()
+        println("Passengers to Sofia - ${passengersToSof}")
+
+        def cap = flights.every {(double)it.passengers / (double)it.capacity >= 0.5}
+        println("Is every flight full at least 50% of its capacity? ${cap ? "Yes" : "No"} ")
+
+        // Is there a flight to London that is running late
+        def londonLate = flights.findAll {it.destination == "LON" }.any { it.runningLate}
+        println("Is there a flight to London running late? ${londonLate ? "Yes" : "No"}")
+
+
+        def pas = flights.findAll { it.runningLate == true}.collect { it.passengers }.sum()
+        println("Passengers running late ${pas}")
+
+        println("All flights to Sofia that are running late:")
+        flights.findAll { it.destination == "SOF" && it.runningLate == true}.each { it.description() }
+
+        println("The first flight that is running late and more than 95% of the capacity is full")
+        flights.findAll { it.runningLate == true }
+                .find { (double)it.passengers / (double)it.capacity > 0.95 }
+                .each { it.description() }
+
+        println("All flights from Sofia")
+        flights.findAll { it.startingLocation == "SOF"}.each { it.description() }
+
+        println("Sorted by day of arrival ascending")
+        flights.sort {a, b -> a.arrivalAt.getDayOfMonth() - b.arrivalAt.getDayOfMonth()}.each {it.description()}
```

Comment:
a.arrivalAt.getDayOfMonth() - b.arrivalAt.getDayOfMonth() 
this is not correct because you can have a 31st of January going to 1st of February.

How can you easily fix this?